### PR TITLE
feat: create an api for the cronjob trigger which cleans up the soft deleted users

### PR DESCRIPTION
**Pull request by sywhb**
- add index for status and updated_at columns on user table
- add a pubsub service endpoint for cleanup soft deleted users
- add test cases
- create omnivore_admin role and allow omnivore_admin to delete all users

### CODE REVIEW

File: packages/api/src/routers/svc/user.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  }
+}
+
+export function userServiceRouter() {
+  const router = express.Router()
+
+  router.post('/prune', cors<express.Request>(corsConfig), async (req, res) => {
+    logger.info('prune soft deleted users')
+
+    const { message: msgStr, expired } = readPushSubscription(req)
```
Comment (jacksonh) [inline packages/api/src/routers/svc/user.ts:40]: This function also validates the token, right?

Comment (sywhb) [reply packages/api/src/routers/svc/user.ts:40]: Yeah

File: packages/api/src/routers/svc/user.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  try {
+    const obj = JSON.parse(msgStr) as unknown
+    if (isCleanupMessage(obj)) {
+      return obj
+    }
+  } catch (err) {
+    console.log('error deserializing event: ', { msgStr, err })
+  }
+
+  return {
```
Comment (jacksonh) [inline packages/api/src/routers/svc/user.ts:29]: Because of this could we sometimes process invalid messages?

Comment (sywhb) [reply packages/api/src/routers/svc/user.ts:29]: Actually, the message here is just to sub the days from the deletedAt timestamp so we can prune users within a period

**Review by jacksonh — APPROVED** [review]
My only fear with this one is we could kind of lock the database up when it happens, imagine if there were many users with large libraries.

**Comment by sywhb** [comment]
> My only fear with this one is we could kind of lock the database up when it happens, imagine if there were many users with large libraries.

Yeah, I have this concern too

**Comment by sywhb** [comment]
I probably should:

1. Rewrite the sql to do batch deletion
2. run the cronjob at off-peak hours every day

**Pull request merged**